src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -0,0 +1,827 @@
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    //TODO: complete me!
+    let liquidityAmounts: { [id: string]: bigint } = {};
+    for (const pool of this.pools) {
+      if (
+        pool.token0 === tokenAddress.toLowerCase() ||
+        pool.token1 === tokenAddress.toLowerCase()
+      ) {
+        const state: FluidDexPoolState = await this.eventPools[
+          pool.id
+        ].getStateOrGenerate(
+          await this.dexHelper.provider.getBlockNumber(),
+          false,
+        );
+
+        liquidityAmounts[pool.id] =
+          pool.token0 === tokenAddress
+            ? state.collateralReserves.token0RealReserves +
+              state.debtReserves.token0RealReserves
+            : state.collateralReserves.token1RealReserves +
+              state.debtReserves.token1RealReserves;
+      }
+    }
+
+    const entries = Object.entries(liquidityAmounts);
+
+    // Sort the entries based on the values in descending order
+    entries.sort((a, b) => {
+      if (b[1] > a[1]) return 1;
+      if (b[1] < a[1]) return -1;
+      return 0;
+    });
+
+    // Take the top k entries
+    const topKEntries = entries.slice(0, limit);
+
+    // Convert the array back to an object
+    const sortedAmounts = Object.fromEntries(topKEntries);
+
+    const poolLiquidities: PoolLiquidity[] = [];
+
+    for (const [id, amount] of Object.entries(sortedAmounts)) {
+      const pool = this.pools.find(p => p.id === id);
+      if (!pool) continue; // Skip if pool not found
+
+      poolLiquidities.push({
+        exchange: 'FluidDex',
+        address: pool.address,
+        connectorTokens: [],
+        liquidityUSD: Number(amount), // converted to number
+      });
+    }
+    return poolLiquidities;
+  }
+
+  // This is optional function in case if your implementation has acquired any resources
+  // you need to release for graceful shutdown. For example, it may be any interval timer
+  releaseResources(): AsyncOrSync<void> {
+    // TODO: complete me!
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: string,
+    destAmount: string,
+    recipient: Address,
+    data: FluidDexData,
+    side: SwapSide,
+    context: Context,
+    executorAddress: Address,
+  ): AsyncOrSync<DexExchangeParam> {
+    let method: string;
+    let args: any;
+    let returnAmountPos: number | undefined = undefined;
+
+    const deadline = getLocalDeadlineAsFriendlyPlaceholder();
+
+    method = 'swapIn';
+    returnAmountPos = extractReturnAmountPosition(
+      this.iFluidDexPool,
+      method,
+      'amountOut_',
+      1,
+    );
+
+    const pool = this.getPoolByTokenPairAddress(srcToken, destToken);
+
+    if (srcToken == '0xEeeeeEeeeeeeEeeeeeeEeeeeEeeeeeEeeeeEeeeE') {
+      throw new Error(
+        'srcToken can not be Native ETH, try exchanging tokens and swapSide',
+      );
+    }
+
+    if (side == SwapSide.SELL) {
```

Comment:
from what I understand
-  SwapSide.BUY would be: define an output amount -> swap. in our case `swapOut` method. 
-  SwapSide.SELL would be: define an input amount -> swap. in our case  `swapIn` method.

We can skip SwapSide.BUY all together for now if it causes any issues. If it works, rather then reverting with throw new Error above, can we simply return that no swap is available for this route?

Reply:
I have changed the code to either return null value or empty type structs. This will not stop the code execution by giving error.

Reply:
throw new Error(
        'srcToken can not be Native ETH, try exchanging tokens and swapSide',
      ); 
      
      still exists? Can you check what other projects do in such a case, return nothing vs revert and adjust accordingly?

Reply:
I see Curve-V2 also reverting on BUY operation, so I have also written the revert statement.